File: 12-Searching-Algorithms/01-Linear-Search/linear_search.py
n = int(input("Enter number of elements: "))
arr = list(map(int, input("Enter elements: ").split()))

target = int(input("Enter element to search: "))

found = -1

for i in range(n):
    if arr[i] == target:
        found = i
        break

if found != -1:
    print("Element found at index:", found)
else:
    print("Element not found.")

File: 12-Searching-Algorithms/01-Linear-Search/linear_search.c
#include <stdio.h>

#define MAX 100

int main() {
    int arr[MAX], n, target;
    int found = -1;

    printf("Enter number of elements: ");
    scanf("%d", &n);

    printf("Enter elements: ");

    for(int i = 0; i < n; i++)
        scanf("%d", &arr[i]);

    printf("Enter element to search: ");
    scanf("%d", &target);

    for(int i = 0; i < n; i++) {
        if(arr[i] == target) {
            found = i;
            break;
        }
    }

    if(found != -1)
        printf("Element found at index: %d\n", found);
    else
        printf("Element not found.\n");

    return 0;
}